<!-- 
  Wrapper launch file for the Motoman specific robot_state node.
-->
<launch>
	<!-- IP of robot (or PC running simulation) -->
	<arg name="robot_ip" />

	<!-- Load the byte-swapping version of robot_state if required -->
	<arg name="use_bswap" />

    <arg name="joint_ns" default = ""/>

	<!-- put them on the parameter server -->
	<param name="robot_ip_address" type="str" value="$(arg robot_ip)" />

	<!-- load the correct version of the robot state node -->
	<node if="$(arg use_bswap)" name="joint_state" 
		  pkg="motoman_driver" type="robot_state_bswap" output="screen">
      <remap from="dx200/dx200_r1_controller/joint_states" to="$(arg joint_ns)/dx200/dx200_r1_controller/joint_states" if="$(eval joint_ns != '')" />
      <remap from="dx200/dx200_s1_controller/joint_states" to="$(arg joint_ns)/dx200/dx200_s1_controller/joint_states" if="$(eval joint_ns != '')"/>
      <remap from="dx200/dx200_r1_controller/feedback_states" to="$(arg joint_ns)/dx200/dx200_r1_controller/feedback_states" if="$(eval joint_ns != '')" />
      <remap from="dx200/dx200_s1_controller/feedback_states" to="$(arg joint_ns)/dx200/dx200_s1_controller/feedback_states" if="$(eval joint_ns != '')"/>
    </node>

	<node unless="$(arg use_bswap)" name="joint_state" 
		  pkg="motoman_driver" type="robot_state" output="screen">
      <remap from="dx200/dx200_r1_controller/joint_states" to="$(arg joint_ns)/dx200/dx200_r1_controller/joint_states" if="$(eval joint_ns != '')" />
      <remap from="dx200/dx200_s1_controller/joint_states" to="$(arg joint_ns)/dx200/dx200_s1_controller/joint_states" if="$(eval joint_ns != '')"/>
      <remap from="joint_states" to="$(arg joint_ns)/joint_states" if="$(eval joint_ns != '')" />
      <remap from="dx200/dx200_r1_controller/feedback_states" to="$(arg joint_ns)/dx200/dx200_r1_controller/feedback_states" if="$(eval joint_ns != '')" />
      <remap from="dx200/dx200_s1_controller/feedback_states" to="$(arg joint_ns)/dx200/dx200_s1_controller/feedback_states" if="$(eval joint_ns != '')"/>
      <remap from="feedback_states" to="$(arg joint_ns)/feedback_states" if="$(eval joint_ns != '')"/>
    </node>

</launch>
